modules/tinymce/src/core/test/ts/browser/selection/GetSelectionContentTest.ts

```diff
@@ -63,45 +66,55 @@ UnitTest.asynctest('browser.tinymce.selection.GetSelectionContentTest', (success
         tinyApis.sSetContent('<p>a</p>'),
         tinyApis.sSetSelection([ 0, 0 ], 0, [ 0, 0 ], 1),
         sAssertGetContent('Should be some content', editor, 'a')
-      ])),
-      Logger.t('Should be text contents provided by override handler', GeneralSteps.sequence([
+      ]),
+      Log.stepsAsStep('TBA', 'Should be text contents provided by override handler', [
         tinyApis.sSetContent('<p>a</p>'),
         tinyApis.sSetSelection([ 0, 0 ], 0, [ 0, 0 ], 1),
         sAssertGetContentOverrideBeforeGetContent('Should be overridden content', editor, 'X')
-      ])),
-      Logger.t(`Should be text contents when editor isn't focused and format is text`, GeneralSteps.sequence([
+      ]),
+      Log.stepsAsStep('TBA', `Should be text contents when editor isn't focused and format is text`, [
         sAddTestDiv,
         tinyApis.sSetContent('<p>ab</p>'),
         tinyApis.sSetSelection([ 0, 0 ], 0, [ 0, 0 ], 2),
         sFocusDiv,
         sAssertGetContent('Should be some content', editor, 'ab', { format: 'text' }),
         sRemoveTestDiv
-      ])),
-      Logger.t('Should be text content with newline', GeneralSteps.sequence([
+      ]),
+      Log.stepsAsStep('TBA', 'Should be text content with newline', [
         tinyApis.sSetContent('<p>ab<br/>cd</p>'),
         tinyApis.sSetSelection([ 0, 0 ], 0, [ 0, 2 ], 2),
         sAssertGetContent('Should be some content', editor, `ab${Env.ie === 11 ? '\r\n' : '\n'}cd`, { format: 'text' })
-      ])),
-      Logger.t('Should be text content with leading visible spaces', GeneralSteps.sequence([
+      ]),
+      Log.stepsAsStep('TBA', 'Should be text content with leading visible spaces', [
         tinyApis.sSetContent('<p>content<em> Leading space</em></p>'),
         tinyApis.sSetSelection([ 0 ], 1, [ 0 ], 2),
         sAssertGetContent('Should be some content', editor, ' Leading space', { format: 'text' })
-      ])),
-      Logger.t('Should be text content with trailing visible spaces', GeneralSteps.sequence([
+      ]),
+      Log.stepsAsStep('TBA', 'Should be text content with trailing visible spaces', [
         tinyApis.sSetContent('<p><em>Trailing space </em>content</p>'),
         tinyApis.sSetSelection([ 0 ], 0, [ 0 ], 1),
         sAssertGetContent('Should be some content', editor, 'Trailing space ', { format: 'text' })
-      ])),
-      Logger.t('Should be text content without non-visible leading/trailing spaces', GeneralSteps.sequence([
+      ]),
+      Log.stepsAsStep('TINY-6448', 'pre blocks should have preserved spaces', [
+        tinyApis.sSetContent('<pre>          This      Has\n     Spaces</pre>'),
+        tinyApis.sSetSelection([ ], 0, [ ], 1),
+        sAssertGetContent('Should be some content', editor, '          This      Has\n     Spaces', { format: 'text' })
+      ]),
+      Log.stepsAsStep('TINY-6448', 'pre blocks should have preserved spaces', [
+        tinyApis.sSetContent('<pre><p>          This      Has\n     Spaces<p></pre>'),
```

Comment:
This is not valid HTML, as a `p` can't be inside a `pre` element. See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/pre. As such TinyMCE will be converting this to:
```
<pre></pre>
<p>          This      Has\n     Spaces<p>
```
which in turn will cause the `pre` to be removed since it's empty. So this won't be testing what you want and also shows there maybe a bug in the changes.